app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,63 @@
+<div class="container">
+  <p style="color: green"><%= notice %></p>
+
+  <h3><%= @recipe.name %></h3>
+
+  <div class="d-flex justify-content-between">
+    <p>Preparation time: <b><%= @recipe.preparation_time %> min</b></p>
+    <%= form_with model: @recipe do |form| %>
+      <div class="form-check form-switch form-check-reverse">
+        <%= form.label :public, class: 'form-check-label' %>
+        <%= form.check_box :public, class: 'form-check-input', role: 'switch', onchange: "this.form.requestSubmit()", disabled: !can?(:manage, @recipe) %>
+      </div>
+    <% end %>
+  </div>
+
+  <p>Cooking time: <b><%= @recipe.cooking_time %> min</b></p>
+  <p><%= @recipe.description %></p>
+
+    <div class="d-flex justify-content-between">
+      <%= link_to 'Generate shopping list', shopping_path(@recipe), class: 'btn btn-primary' %>
+      <% if can? :manage, @recipe %>
+        <%= link_to "Add ingredients", edit_recipe_path(@recipe), class: 'btn btn-primary' %>
```

Comment:
- [ ]  When I click on the  link to Generate shopping list I  see  this error because if I add an ingredient and don't add `Quantity` the error is shown, Kindly add validation to the input so an input  is not empty see the images below ⬇️ 



![Screenshot_178](https://github.com/andrianarivo/recipe_app/assets/69638013/dce4a57b-6163-484b-b6c8-2d1c4d2795c6)



https://github.com/andrianarivo/recipe_app/assets/69638013/d8438d20-c2ae-4554-a535-5fbff00c941e

Reply:
Thank you @prolajumokeoni for your thorough review, we'll make the necessary changes and notify you soon